Storage/SQL/SQLiteLogins.swift

```diff
@@ -249,6 +249,37 @@ public class SQLiteLogins: BrowserLogins {
         return db.runQuery(sql, args: args, factory: SQLiteLogins.LoginDataFactory)
     }
 
+    public func getAllLogins() -> Deferred<Maybe<Cursor<LoginData>>> {
+        return searchLoginsByUsername(nil)
+    }
+
+    public func searchLoginsByUsername(username: String?, orPassword password: String? = nil, orHostname hostname: String? = nil) -> Deferred<Maybe<Cursor<LoginData>>> {
+        let projection = SQLiteLogins.MainWithLastUsedColumns
+        var searchClauses = [String]()
+        if let username = username {
+            searchClauses.append(" username LIKE '\(username)%' ")
```

Comment:
Ol' Bobby Tables is going to have fun here, let alone folks with quotes in their passwords…